### ISSUE TITLE
[Bug]: Gas - `TypeError: Cannot set properties of undefined (setting 'gasFeeEstimates')`

### ISSUE THREAD

**Issue by seaona** [MetaMask/metamask-extension#27241]
### Describe the bug

Background error `TypeError: Cannot set properties of undefined (setting 'gasFeeEstimates')`. I haven't seen any UI impact, but cannot reproduce again, to be able to add more details

### Screenshots/Recordings

![Screenshot from 2024-09-18 10-59-46](https://github.com/user-attachments/assets/611549eb-70f7-433f-89ee-de5d45b9069e)

### Steps to reproduce

To determine.

I was using 2 networks for 2 different dapps (Sepolia and Linea Sepolia) and switching back and forth triggering txs.

### Detection stage

During release testing

### Version

12.3.0

### Build type

None

### Browser

Chrome

### Operating system

Linux

**Comment by cryptotavares** [comment]
@seaona if you cannot reproduce should we close it?

**Comment by seaona** [comment]
I don't think we should close it. The issue is still present, just encountered again in 12.3.0 and I've seen that [there are more than 90k entries](https://metamask.sentry.io/issues/5299136537/?project=273505&query=gasfeeestimates&referrer=issue-stream&sort=freq&statsPeriod=90d&stream_index=0) for this error in Sentry. So addressing it would definitely help reducing Sentry errors

![Screenshot from 2024-10-03 11-42-05](https://github.com/user-attachments/assets/9abc936b-21f1-493a-b277-fb8e2d2ed5b6)

What do you think?

![Screenshot from 2024-10-03 11-40-02](https://github.com/user-attachments/assets/629738b0-2e34-43c5-973a-f58466d9b424)

**Comment by seaona** [comment]
Found some repro steps cc @cryptotavares :

**Repro:**
1. Setup proxy (ie Burpsuite)
2. Intercept the `gas.api.cx.metamask.io` requests
3. Start a Send flow
4. Leave all the gas API requests unresolved
5. Go to the last confirmation screen
6. Wait a couple of seconds
7. Open the Edit gas
8. See how gas is trying to refresh
9. Confirm tx
10. See background error

https://github.com/user-attachments/assets/569bfaef-6b14-419f-93f3-d929e2124c1a

**Comment by vinistevam** [comment]
The issue will be fixed by [PR](https://github.com/MetaMask/metamask-extension/pull/29312).
